File: Top 100 Codes/05-Operations on Strings/16-Find non-repeating characters in a string/Findnonrepeatingcharactersinastring.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

void findNonRepeatingCharacters(const string& str) {
    unordered_map<char, int> freq;

    // Count the frequency of each character 
    for (int i = 0; i < str.length(); i++) {
        freq[str[i]]++;
    }

    cout << "Non-repeating characters: ";
    bool found = false;
    for (int i = 0; i < str.length(); i++) {
        if (freq[str[i]] == 1) {
            cout << str[i] << " ";
            found = true;
        }
    }

    if (!found) {
        cout << "None";
    }
}

int main() {
    string input;
    cout << "Enter a string: ";
    getline(cin, input);

    findNonRepeatingCharacters(input);

    return 0;
}

File: Top 100 Codes/05-Operations on Strings/16-Find non-repeating characters in a string/Findnonrepeatingcharactersinastring.py
def find_non_repeating_characters(s):
    freq = {}

    # Count frequency of each character
    for i in range(len(s)):
        char = s[i]
        freq[char] = freq.get(char, 0) + 1

    print("Non-repeating characters:", end=" ")
    found = False
    for i in range(len(s)):
        if freq[s[i]] == 1:
            print(s[i], end=" ")
            found = True

    if not found:
        print("None")

# Input
input_str = input("Enter a string: ")
find_non_repeating_characters(input_str)
